modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4SizeHeaderFrameDecoder.java

```diff
@@ -33,10 +33,19 @@
     @Override
     protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) throws Exception {
         try {
-            boolean continueProcessing = TcpTransport.validateMessageHeader(Netty4Utils.toBytesReference(in));
-            final ByteBuf message = in.skipBytes(TcpHeader.MARKER_BYTES_SIZE + TcpHeader.MESSAGE_LENGTH_SIZE);
-            if (!continueProcessing) return;
-            out.add(message);
+            final ByteBufStreamInput input = new ByteBufStreamInput(in, in.readableBytes());
+            int size = TcpTransport.validateMessageHeader(input);
+            if (size == -1) { // that's a ping - we just ignore it.
```

Comment:
`-1` -> TcpTransport.PING_DATA_SIZE so that we can easily find where it's set and use?